streampipes-extensions/streampipes-sinks-databases-jvm/src/main/java/org/apache/streampipes/sinks/databases/jvm/milvus/MilvusSink.java

```diff
@@ -0,0 +1,380 @@
+import org.apache.streampipes.model.graph.DataSinkDescription;
+import org.apache.streampipes.model.runtime.Event;
+import org.apache.streampipes.model.schema.EventProperty;
+import org.apache.streampipes.model.schema.EventPropertyNested;
+import org.apache.streampipes.model.schema.EventPropertyPrimitive;
+import org.apache.streampipes.model.schema.EventSchema;
+import org.apache.streampipes.model.schema.PropertyScope;
+import org.apache.streampipes.sdk.builder.DataSinkBuilder;
+import org.apache.streampipes.sdk.builder.StreamRequirementsBuilder;
+import org.apache.streampipes.sdk.helpers.EpRequirements;
+import org.apache.streampipes.sdk.helpers.Labels;
+import org.apache.streampipes.sdk.helpers.Locales;
+import org.apache.streampipes.sdk.helpers.Options;
+import org.apache.streampipes.wrapper.params.compat.SinkParams;
+import org.apache.streampipes.wrapper.standalone.StreamPipesDataSink;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import io.milvus.param.Constant;
+import io.milvus.pool.MilvusClientV2Pool;
+import io.milvus.pool.PoolConfig;
+import io.milvus.v2.client.ConnectConfig;
+import io.milvus.v2.client.MilvusClientV2;
+import io.milvus.v2.common.DataType;
+import io.milvus.v2.common.IndexParam;
+import io.milvus.v2.service.collection.request.AddFieldReq;
+import io.milvus.v2.service.collection.request.CreateCollectionReq;
+import io.milvus.v2.service.collection.request.DescribeCollectionReq;
+import io.milvus.v2.service.collection.request.HasCollectionReq;
+import io.milvus.v2.service.collection.response.DescribeCollectionResp;
+import io.milvus.v2.service.database.request.CreateDatabaseReq;
+import io.milvus.v2.service.vector.request.InsertReq;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MilvusSink extends StreamPipesDataSink {
+  public static final String MILVUS_URI_KEY = "milvus_uri";
+  public static final String MILVUS_TOKEN_KEY = "milvus_token";
+
+  public static final String MILVUS_DBNAME_KEY = "milvus_dbname";
+  public static final String DATABASE_REPLICA_NUMBER_KEY = "database_replica_number";
+
+  public static final String COLLECTION_NAME_KEY = "collection_name";
+
+  public static final String VECTOR_KEY = "vector";
+  public static final String DIMENSION = "dimension";
+  public static final String METRIC_TYPE = "metric_type";
+
+  public static final String INDEX = "index";
+
+  public static final String PRIMARY = "primary";
+
+  public DataType vectorDataType;
+
+  public final Gson gson = new Gson();
+
+  public MilvusClientV2Pool pool;
+  public MilvusClientV2 client;
+  String vector;
+  String primary;
+  String collectionName;
+  Integer dimension;
+  IndexParam.MetricType metricType;
+  IndexParam indexParam;
+  CreateCollectionReq.CollectionSchema collectionSchema;
+
+  public static final String BYTE = "http://www.w3.org/2001/XMLSchema#byte";
```

Comment:
There is an `Xsd` class in `org.apache.streampipes.vocabulary` which provide these variables, could you reuse these?